### PR TITLE
Update Documentation for Data Type

### PR DESCRIPTION
**Pull request by jinimukh**
This PR adds a page under `Advanced` to detail the different data types Lux uses. It also links to a tutorial in lux-binder.

### CODE REVIEW

File: doc/source/advanced/datatype.rst [hunk 1]
```diff
@@ -0,0 +1,90 @@
+***********************************
+Data Types in Lux
+***********************************
+
+.. note:: You can follow along this tutorial in a Jupyter notebook. 
+
+Currently, Lux supports four data types:
+
+* Quantitative
```
Comment (dorisjlee) [inline doc/source/advanced/datatype.rst:9]: Change this to a linked table of contents via `.. contents:: :local:`

File: doc/source/advanced/datatype.rst [hunk 2]
```diff
@@ -0,0 +1,90 @@
+***********************************
+Data Types in Lux
```
Comment (dorisjlee) [inline doc/source/advanced/datatype.rst:2]: `Defining Data Type`, no need `in Lux` since it's clear what we're referring to here.

File: doc/source/advanced/datatype.rst [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+***********************************
+
+.. note:: You can follow along this tutorial in a Jupyter notebook.
```
Comment (dorisjlee) [inline doc/source/advanced/datatype.rst:5]: Add link to `[Github] [Binder]` here

File: doc/source/advanced/datatype.rst [hunk 4]
```diff
@@ -0,0 +1,90 @@
+***********************************
+Data Types in Lux
+***********************************
+
+.. note:: You can follow along this tutorial in a Jupyter notebook. 
+
+Currently, Lux supports four data types:
```
Comment (dorisjlee) [inline doc/source/advanced/datatype.rst:7]: This is a bit abrupt, start by introducing what are data types and why are data types important in Lux.

File: doc/source/advanced/datatype.rst [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+* Nominal
+* ID
+
+Note that these data types are different from Pandas `dtype`. These data types are automatically inferred based on the `dataframe` inputted.
+The following code snippet shows what data types were inferred:
+
+.. code-block:: python
+
+  df = pd.read_csv("https://github.com/lux-org/lux-datasets/blob/master/data/college.csv?raw=true")
+  df.data_type
```
Comment (dorisjlee) [inline doc/source/advanced/datatype.rst:20]: display the content of the output cell here

File: doc/source/advanced/datatype.rst [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+
+Quantitative
+------------
+The Quantitative data type is used when there is a count or measure of a certain attribute.
```
Comment (dorisjlee) [inline doc/source/advanced/datatype.rst:25]: I think we should first say that something is quantitative if it is a numerical column. Measure/counts are examples of quantitative but they don't define it.

File: doc/source/advanced/datatype.rst [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+.. image:: https://github.com/jinimukh/lux-resources/blob/datatype/doc_img/datatype-3.png?raw=true
+  :width: 700
+  :align: center
+  :alt: Displays line graph for temporal variables.
+
+ID
+---
+The ID data type is chosen for any column that looks like an ID and shouldn't be plotted. For example, zip code, user ID, etc.
+
```
Comment (dorisjlee) [inline doc/source/advanced/datatype.rst:72]: add example of ID and show the warning message of how it is not plotted

File: doc/source/advanced/datatype.rst [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+Here, specifying intent as the temporal variable, the `Temporal` tab reflects a line graph of the records and time.
+
+.. image:: https://github.com/jinimukh/lux-resources/blob/datatype/doc_img/datatype-3.png?raw=true
+  :width: 700
+  :align: center
+  :alt: Displays line graph for temporal variables.
+
+ID
+---
+The ID data type is chosen for any column that looks like an ID and shouldn't be plotted. For example, zip code, user ID, etc.
```
Comment (dorisjlee) [inline doc/source/advanced/datatype.rst:71]: Elaborate on ID description a bit more.

File: doc/source/advanced/datatype.rst [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+-------------------------------
+Sometimes, Lux incorrectly identifies the correct data. Here is an example:
+
+.. code-block:: python
+
+    df = pd.read_csv("https://github.com/lux-org/lux-datasets/blob/master/data/real_estate_tutorial.csv?raw=true")
+    df
+
+
+Lux incorrectly assumes that this is a temporal column because of the column name. However, in reality, the columns are an ordered months and years. Here is one way to fix the problem:
```
Comment (dorisjlee) [inline doc/source/advanced/datatype.rst:84]: Unclear which column `this is` referring to. We should also add a snippet of what the real estate dataset looks like for the non-notebook documentation.

File: doc/source/advanced/datatype.rst (hunk 7 above)
Comment (dorisjlee) [reply doc/source/advanced/datatype.rst:72]: Explain how we correct if something is mislabelled as ID when it shouldn't be, and vice versa.

File: doc/source/advanced/datatype.rst [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+The ID data type is chosen for any column that looks like an ID and shouldn't be plotted. For example, zip code, user ID, etc.
+
+
+Changing the Inferred Data Type
+-------------------------------
+Sometimes, Lux incorrectly identifies the correct data. Here is an example:
+
+.. code-block:: python
+
+    df = pd.read_csv("https://github.com/lux-org/lux-datasets/blob/master/data/real_estate_tutorial.csv?raw=true")
```
Comment (dorisjlee) [inline doc/source/advanced/datatype.rst:80]: This example is not super clear, because technically these are temporal columns. Is there another example where Lux makes a mistake when there is ambiguity in the data type? e.g.) ID fields that should not be labelled as IDs or ordinal data labeled as nominal data.

File: doc/source/advanced/datatype.rst [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+  :align: center
+  :alt: Displays correlation of quantitative variables.
+
+Nominal
+--------
+The Nominal data type is for categorical data. 
+For example, `PredominantDegree` is nominal because rather than being an explicit measure, it describes an attribute. 
+In this case, there are three possible values: `Associate`, `Bachelor's`, and `Certificate`.
+Lux displays these variables under the `Occurrence` tab as bar charts for the number of occurrences. 
+Upon more granular searches by specifying intent, stacked bar charts comparing two nominal variables is also possible.
```
Comment (dorisjlee) [inline doc/source/advanced/datatype.rst:41]: Unclear what this sentence means.

File: doc/source/advanced/datatype.rst [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+
+.. image:: https://github.com/jinimukh/lux-resources/blob/datatype/doc_img/datatype-2.png?raw=true
+  :width: 700
+  :align: center
+  :alt: Displays bar chart for nominal variables.
```
Comment (dorisjlee) [inline doc/source/advanced/datatype.rst:46]: Can we add info about `ordinal` data types too? We should add support for this also.

**Review by dorisjlee — CHANGES_REQUESTED** [review]

File: doc/source/advanced/datatype.rst [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+
+Temporal
+--------
+The temporal data type is used when Lux thinks based on either the format of the data passed in or the title of the column that the data in that column is time-sensitive.
```
Comment (dorisjlee) [inline doc/source/advanced/datatype.rst:50]: `time-sensitive`--> `time-related`

**Review by dorisjlee — CHANGES_REQUESTED** [review]

File: doc/source/advanced/datatype.rst (hunk 2 above)
Comment (jinimukh) [reply doc/source/advanced/datatype.rst:2]: Ahh I wanted to specify in Lux so that its clear from the get-go that this page is about Lux Data Types and not dtype

File: doc/source/advanced/datatype.rst (hunk 3 above)
Comment (jinimukh) [reply doc/source/advanced/datatype.rst:5]: I can add this once the tutorial is merged!

File: doc/source/advanced/datatype.rst (hunk 12 above)
Comment (jinimukh) [reply doc/source/advanced/datatype.rst:46]: Technically, we don't support ordinal data type, I believe. I'll add this when we add support?

File: doc/source/advanced/datatype.rst (hunk 4 above)
Comment (thyneb19) [reply doc/source/advanced/datatype.rst:7]: This seems like a good spot to give a brief overview on how Lux uses these data types for chart generation.

File: doc/source/advanced/datatype.rst [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+  df = pd.read_csv("https://github.com/lux-org/lux-datasets/blob/master/data/college.csv?raw=true")
+  df.data_type
+
+
+Quantitative
+------------
+The quantitative data type is used when there is a count or measure of a certain attribute. 
+In the example above, the column `AcceptanceRate` is quantitative because it is a measure. 
+Also, any aggregate such as means and medians will be categorized as quantitative. 
+Usually, Lux is able to compute Correlations between two quantitative columns or show the distribution of each column.
```
Comment (thyneb19) [inline doc/source/advanced/datatype.rst:28]: Section looks good! Maybe you could also mention how correlations are shown via scatterplots/heatmaps and distributions via histograms

File: doc/source/advanced/datatype.rst (hunk 11 above)
Comment (thyneb19) [reply doc/source/advanced/datatype.rst:41]: Maybe rephrase to something like "When two nominal variables are specified in an intent, a stacked bar chart is used to compare the two variables"?

Comment (thyneb19) [reply doc/source/advanced/datatype.rst:41]: On a separate note on documentation, @dorisjlee do we have a section somewhere outlining all of the supported chart types in Lux?

File: doc/source/advanced/datatype.rst [hunk 15]
```diff
@@ -1,31 +1,44 @@
...
 
+.. image:: https://github.com/jinimukh/lux-resources/blob/datatype/doc_img/datatype-4.png?raw=true
+  :width: 300
+  :align: center
 
 Quantitative
 ------------
-The quantitative data type is used when there is a count or measure of a certain attribute. 
+Quantitative data is used to describe numerical data. 
+The is usually used when Lux detects that there is a count or measure of a certain attribute.
```
Comment (thyneb19) [inline doc/source/advanced/datatype.rst:38]: Small typo at the start of this line, maybe adjust it to read something like:
"This data type is typically assigned when Lux detects that there is a count or measure of a certain attribute."

File: doc/source/advanced/datatype.rst [hunk 16]
```diff
@@ -1,31 +1,44 @@
...
 
 Quantitative
 ------------
-The quantitative data type is used when there is a count or measure of a certain attribute. 
+Quantitative data is used to describe numerical data. 
+The is usually used when Lux detects that there is a count or measure of a certain attribute. 
 In the example above, the column `AcceptanceRate` is quantitative because it is a measure. 
 Also, any aggregate such as means and medians will be categorized as quantitative. 
-Usually, Lux is able to compute Correlations between two quantitative columns or show the distribution of each column.
+Usually, Lux is able to compute Correlations between two quantitative columns and display them via scatterplots/heatmaps.
```
Comment (thyneb19) [inline doc/source/advanced/datatype.rst:41]: I think we can remove the "usually" from the beginning of this line.

**Review by thyneb19 — CHANGES_REQUESTED** [review]
Looks good, thanks Ujjaini! Added some comments pointing out some small typos but once that's fixed I think the rest looks great.

**Review by thyneb19 — APPROVED** [review]

File: doc/source/advanced/datatype.rst (hunk 3 above)
Comment (dorisjlee) [reply doc/source/advanced/datatype.rst]: I merged the tutorials in now, you can add in the links to Github and Binder

File: doc/source/advanced/datatype.rst (hunk 12 above)
Comment (dorisjlee) [reply doc/source/advanced/datatype.rst:55]: Yeah, let's defer this to later

**Review by dorisjlee — APPROVED** [review]

**Comment by dorisjlee** [comment]
Thanks @jinimukh for this comprehensive data type tutorial! And for @thyneb19 for looking over this!
